WordPress/Classes/ViewRelated/Menus/MenuItemsViewController.m

```diff
@@ -0,0 +1,866 @@
+    toggledItemView.showsEditingButtonOptions = NO;
+    
+    [self addNewInsertionViewWithOrder:MenuItemInsertionOrderAbove forItemView:toggledItemView];
+    [self addNewInsertionViewWithOrder:MenuItemInsertionOrderBelow forItemView:toggledItemView];
+    [self addNewInsertionViewWithOrder:MenuItemInsertionOrderChild forItemView:toggledItemView];
+}
+
+- (void)insertItemInsertionViewsAroundItemView:(MenuItemView *)toggledItemView animated:(BOOL)animated
+{
+    BOOL wasEditing = self.isEditingForItemViewInsertion;
+
+    CGRect previousRect = toggledItemView.frame;
+    CGRect updatedRect = toggledItemView.frame;
+    
+    [self insertInsertionItemViewsAroundItemView:toggledItemView];
+    
+    if (!animated) {
+        return;
+    }
+    
+    // since we are adding content above the toggledItemView, the toggledItemView (focus) will move downwards with the updated content size
+    updatedRect.origin.y += MenuItemsStackableViewDefaultHeight;
+    
+    for (MenuItemInsertionView *insertionView in self.insertionViews) {
+        insertionView.hidden = YES;
+        insertionView.alpha = 0.0;
+    }
+    [UIView animateWithDuration:0.3 animations:^{
+        
+        for (MenuItemInsertionView *insertionView in self.insertionViews) {
+            insertionView.hidden = NO;
+            insertionView.alpha = 1.0;
+        }
+        // inform the delegate to handle this content change based on the rect we are focused on
+        // a delegate will likely scroll the content with the size change
+        if (!wasEditing) {
+            [self.delegate itemsViewAnimatingContentSizeChanges:self focusedRect:previousRect updatedFocusRect:updatedRect];
+        }
+        
+    } completion:^(BOOL finished) {
+        
+        
+    }];
+}
+
+- (void)removeItemInsertionViews
+{
+    for (MenuItemInsertionView *insertionView in self.insertionViews) {
+        [self.stackView removeArrangedSubview:insertionView];
+        [insertionView removeFromSuperview];
+    }
+    
+    [self.insertionViews removeAllObjects];
+    self.itemViewForInsertionToggling = nil;
+}
+
+- (void)removeItemInsertionViews:(BOOL)animated
+{
+    self.isEditingForItemViewInsertion = NO;
+    self.itemViewForInsertionToggling.showsCancelButtonOption = NO;
+    self.itemViewForInsertionToggling.showsEditingButtonOptions = YES;
+    
+    if (!animated) {
+        [self removeItemInsertionViews];
+        return;
+    }
+    
+    CGRect previousRect = self.itemViewForInsertionToggling.frame;
+    CGRect updatedRect = previousRect;
+    // since we are removing content above the toggledItemView, the toggledItemView (focus) will move upwards with the updated content size
+    updatedRect.origin.y -= MenuItemsStackableViewDefaultHeight;
+    
+    [UIView animateWithDuration:0.3 delay:0.0 options:0 animations:^{
+        
+        for (MenuItemInsertionView *insertionView in self.insertionViews) {
+            insertionView.hidden = YES;
+            insertionView.alpha = 0.0;
```

Comment:
(Same as above) maybe replace the magic numbers with `WPGUIConstants.h` constants?